File: go/leetcode/50. Pow(x, n).go
func myPow(x float64, n int) float64 {
    if n == 0 {return 1}
    
    partial := myPow(x, int(n / 2))
    
    if n % 2 == 0 { return partial * partial}
    if n < 0 { return (partial * partial) / x}
    return x * partial * partial 
}
 

File: javascript/leetcode/50. Pow(x, n).js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
  if (n === 0) return 1;

  let partial = myPow(x, Number.parseInt(n / 2));

  if (n % 2 === 0) return partial * partial;
  else {
    if (n < 0) return (partial * partial) / x;
    return x * partial * partial;
  }
};
